archaius2-api/src/main/java/com/netflix/archaius/api/PropertyContainer.java

```diff
@@ -85,4 +86,6 @@
      * should be optimized to call one of the known parsing methods based on type. 
      */
     <T> Property<T> asType(Class<T> type, T defaultValue);
+    
+    <T> Property<T> asType(Function<String, T> type, String defaultValue);
```

Comment:
Other than throwing an exception I'm not sure there's any valid default here.  I doubt anyone is implementing this anyway.